src/run-gc.js

```diff
@@ -0,0 +1,48 @@
+const i18n = require('i18next')
+const logger = require('./common/logger')
+const { showDialog, recoverableErrorDialog } = require('./dialogs')
+const dock = require('./dock')
+
+module.exports = function runGarbageCollector ({ getIpfsd }) {
+  dock.run(async () => {
+    logger.info('[run gc] alerting user for effects')
+
+    const opt = showDialog({
+      title: i18n.t('runGarbageCollectorWarning.title'),
+      message: i18n.t('runGarbageCollectorWarning.message'),
+      type: 'warning',
+      buttons: [
+        i18n.t('runGarbageCollectorWarning.action'),
+        i18n.t('cancel')
+      ],
+      showDock: false
+    })
+
+    if (opt !== 0) {
+      logger.info('[run gc] user canceled')
+      return
+    }
+
+    const ipfsd = await getIpfsd()
+
+    if (!ipfsd) {
+      return
+    }
+
+    try {
+      ipfsd.api.repo.gc()
```

Comment:
I know this is still for the old API, but when you switch to the new one in #1411 `repo.gc` will return async iterable which yields an Error or one line each time:
https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/REPO.md#repogc

I think the proper blocking and error handling would be something like:

```js
let error // keep first error, or this could be array too
for await (const res of ipfs.repo.gc()) {
  if (typeof res === 'Error') error = res
}
if (error) throw error
```

@hacdias  mind updating #1411 to  and fixing in a separate PR, if needed?

Reply:
Thanks for the heads up! Wasn't aware of the change on that API.